### PR TITLE
config: warn for unrecognized keys

### PR DESCRIPTION
**Pull request by CapacitorSet**

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -50,6 +50,8 @@ var format = logging.MustStringFormatter(
 
 // Config defines the central type where all configuration is umarhsalled to.
 type Config struct {
+	TomlMetadata toml.MetaData
```
Comment (nl5887) [inline config/config.go]: What about having Config embed toml.MetaData, like this:

```
type Config struct {
toml.MetaData
}
```

Comment (CapacitorSet) [reply config/config.go]: Makes sense, I fixed it.

**Comment by CapacitorSet** [comment]
While I was at it, I also added a commit to warn for unused channel. The first times I used Honeytrap it happened that I added channels and expected them to work "out of the box", and wondering why they didn't - a warning message would be useful.

**Comment by nl5887** [comment]
thanks!

**Pull request merged**